src/cmds/qsub.c

```diff
@@ -4087,13 +4087,16 @@ copy_env_value(char *dest, /* destination  */
 
 			case ESC_CHAR:			/* backslash in value, escape it */
 				*dest++ = *pv;
-				*dest++ = *pv;
+				if (*(pv+1) != ',') /* do not escape if ESC_CHAR already escapes */
+					*dest++ = *pv;
 				break;
 
 			case ',':
 				if (q_ch || quote_flg) {
 					*dest++ = ESC_CHAR;
 					*dest++ = *pv;
+				} else if (*(dest-1) == ESC_CHAR) { /* the comma is escaped, not finished yet */
+					*dest++ = *pv;
```

Comment:
What if the string starts with a comma? We would not want to check the value of *(dest-1) in that case.

Reply:
This should not happen. ",var1='A,B,C,D'" is rejected before it enters copy_env_value(), but you are right, it is better to handle it. Please @mike0042, have a look.